### PR TITLE
Added nano time native call for linux.  Other platforms default to cu…

### PR DESCRIPTION
**Pull request by lwahlmeier**
@ucdseniordesign/owners Take a look at this pull request.  It shows how to add native calls.  There will be more work needed to do this with the javax package and openssl but this is the basics.

This has not been tested on anything other then linux, so please try it on mac/windows if you get a chance and let me know if there are any problems (tests might fail).  

If your feeling adventurous feel free to figure out how to do this on windows and mac and implement it.

### CODE REVIEW

File: classpath/java/lang/System.java [hunk 1]
```diff
@@ -114,9 +110,11 @@ private static Properties makeProperties() {
   public static native long currentTimeMillis();
 
   public static native int identityHashCode(Object o);
+  
+  private static native long getBaseNanoTime();
```
Comment (lwahlmeier) [inline classpath/java/lang/System.java:114]: This is the new native call.  I could have just made the nanoTime() call the native function, but it works either way.

File: classpath/java/util/Properties.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
 import java.io.IOException;
 import java.io.Reader;
 
-public class Properties extends Hashtable {
+public class Properties extends Hashtable<Object, Object> {
```
Comment (lwahlmeier) [inline classpath/java/util/Properties.java:19]: This is an unrelated change, basically this was an uninitialized generic and I just initialized it.  It would have defaulted to Object anyway.

File: test/NanoTimeTest.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import static avian.testing.Asserts.*;
+
+public class NanoTimeTest {
+
+  public static void main(String[] args) throws InterruptedException {
+    long last = System.nanoTime();
+    for(int i=0; i<100; i++) {
+      Thread.sleep(1);
+      long nt = System.nanoTime();
+      assertTrue(nt>last);
```
Comment (lwahlmeier) [inline test/NanoTimeTest.java:11]: It is possible this test could fail on non linux platforms.  It shouldnt, but it has a higher chance of it.

File: classpath/java/lang/System.java (hunk 1 above)
Comment (minor-jason) [inline classpath/java/lang/System.java:114]: Is there a c file for getBaseNanoTime() somewhere?

Comment (lwahlmeier) [inline classpath/java/lang/System.java:114]: its in the pull request.  Its not a file just for it, but its a method the java-lang.cpp file.

Comment (minor-jason) [inline classpath/java/lang/System.java:114]: I see. Thanks!

File: classpath/java-lang.cpp [hunk 4]
```diff
@@ -1057,6 +1057,27 @@ extern "C" JNIEXPORT jlong JNICALL
...
+#else
+#ifdef __APPLE__
+    return Java_java_lang_System_currentTimeMillis(e, jc)*1000*1000;
+#else
+    struct timespec time;
+    int status = clock_gettime(CLOCK_MONOTONIC, &time);
+    if(status == -1) {
+        return Java_java_lang_System_currentTimeMillis(e, jc)*1000*1000;
+    } else {
+        jlong outtime = (jlong(time.tv_sec) * (1000*1000*1000) + jlong(time.tv_nsec));
```
Comment (d2chau) [inline classpath/java-lang.cpp]: nit: Do we need the extra assignment here?

Comment (lwahlmeier) [inline classpath/java-lang.cpp]: Probably not, Feel free to change it.

File: test/NanoTimeTest.java (hunk 3 above)
Comment (d2chau) [inline test/NanoTimeTest.java:11]: I just tested this on OSX, it seems to still work for me.
